File: GO/prime.go
package main

import (
	"fmt"
	"time"
)

// Function to check if a number is prime
func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	if n == 2 || n == 3 {
		return true
	}
	if n%2 == 0 || n%3 == 0 {
		return false
	}
	for i := 5; i*i <= n; i += 6 {
		if n%i == 0 || n%(i+2) == 0 {
			return false
		}
	}
	return true
}

// Function to compute the sum of primes up to a given limit
func sumOfPrimes(limit int) int {
	sum := 2
	for i := 3; i <= limit; i += 2 {
		if isPrime(i) {
			sum += i
		}
	}
	return sum
}

func main() {
	limit := 10_000_000 // Adjust for difficulty
	startTime := time.Now()

	sum := sumOfPrimes(limit)

	duration := time.Since(startTime)
	fmt.Printf("Sum of primes up to %d: %d\n", limit, sum)
	fmt.Printf("Time taken: %v\n", duration)
}

File: C/prime.c
#include <stdio.h>
#include <math.h>
#include <time.h>

// Function to check if a number is prime
int is_prime(int n) {
    if (n < 2) return 0;
    for (int i = 2; i <= sqrt(n); i++) {
        if (n % i == 0) return 0;
    }
    return 1;
}

// Function to compute the sum of primes up to a given limit
long long sum_of_primes(int limit) {
    long long sum = 0;
    for (int i = 2; i <= limit; i++) {
        if (is_prime(i)) {
            sum += i;
        }
    }
    return sum;
}

int main() {
    int limit = 10000000; // Adjust for difficulty
    clock_t start_time = clock();

    long long sum = sum_of_primes(limit);

    double duration = (double)(clock() - start_time) / CLOCKS_PER_SEC;
    printf("Sum of primes up to %d: %lld\n", limit, sum);
    printf("Time taken: %.3f seconds\n", duration);

    return 0;
}
